internal/utils/temporary_utils.go

```diff
@@ -0,0 +1,118 @@
+// TO BE REMOVED AFTER INTEGRATING ADDON INSTANCE CLIENT SDK
```

Comment:
This file is in dire need of some unittests :)

Reply:
Yes! I planned to write the unit tests after the implementation was accepted for this status reporting. Otherwise, after every review, the tests would have required to get refactored as well :)